flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/api/config/ExecutionConfigOptions.java

```diff
@@ -83,97 +83,97 @@
 
-	public static final ConfigOption<Integer> SQL_RESOURCE_EXTERNAL_BUFFER_MEM =
-			key("sql.resource.external-buffer.memory.mb")
-					.defaultValue(10)
+	public static final ConfigOption<String> TABLE_EXEC_RESOURCE_EXTERNAL_BUFFER_MEM =
+			key("table.exec.resource.external-buffer.memory")
+					.defaultValue("10 mb")
 					.withDescription("Sets the externalBuffer memory size that is used in sortMergeJoin and overWindow.");
 
-	public static final ConfigOption<Integer> SQL_RESOURCE_HASH_AGG_TABLE_MEM =
-			key("sql.resource.hash-agg.table.memory.mb")
-					.defaultValue(128)
-					.withDescription("Sets the table memory size of hashAgg operator.");
+	public static final ConfigOption<String> TABLE_EXEC_RESOURCE_HASH_AGG_TABLE_MEM =
+			key("table.exec.resource.hash-agg.memory")
+					.defaultValue("128 mb")
+					.withDescription("Sets the managed memory size of hash aggregate operator.");
 
-	public static final ConfigOption<Integer> SQL_RESOURCE_HASH_JOIN_TABLE_MEM =
-			key("sql.resource.hash-join.table.memory.mb")
-					.defaultValue(128)
-					.withDescription("Sets the HashTable reserved memory for hashJoin operator. It defines the lower limit.");
+	public static final ConfigOption<String> TABLE_EXEC_RESOURCE_HASH_JOIN_TABLE_MEM =
+			key("table.exec.resource.hash-join.memory")
+					.defaultValue("128 mb")
+					.withDescription("Sets the managed memory for hash join operator. It defines the lower limit.");
 
-	public static final ConfigOption<Integer> SQL_RESOURCE_SORT_BUFFER_MEM =
-			key("sql.resource.sort.buffer.memory.mb")
-					.defaultValue(128)
+	public static final ConfigOption<String> TABLE_EXEC_RESOURCE_SORT_BUFFER_MEM =
+			key("table.exec.resource.sort.buffer.memory")
+					.defaultValue("128 mb")
 					.withDescription("Sets the buffer memory size for sort.");
 
 	// ------------------------------------------------------------------------
 	//  Agg Options
 	// ------------------------------------------------------------------------
 
-	public static final ConfigOption<Integer> SQL_EXEC_WINDOW_AGG_BUFFER_SIZE_LIMIT =
-			key("sql.exec.window-agg.buffer-size-limit")
+	public static final ConfigOption<Integer> TABLE_EXEC_WINDOW_AGG_BUFFER_SIZE_LIMIT =
+			key("table.exec.window-agg.buffer-size-limit")
 					.defaultValue(100 * 1000)
 					.withDescription("Sets the window elements buffer size limit used in group window agg operator.");
 
 	// ------------------------------------------------------------------------
 	//  Async Lookup Options
 	// ------------------------------------------------------------------------
-	public static final ConfigOption<Integer> SQL_EXEC_LOOKUP_ASYNC_BUFFER_CAPACITY =
-			key("sql.exec.lookup.async.buffer-capacity")
+	public static final ConfigOption<Integer> TABLE_EXEC_LOOKUP_ASYNC_BUFFER_CAPACITY =
+			key("table.exec.lookup.async.buffer-capacity")
```

Comment:
`table.exec.lookup.async-buffer-capacity` or just `table.exec.lookup.buffer-capacity`?

Reply:
Currently, we have two implementation for lookup a dimension table, one is synchronically, the other is asynchronously. So I think `async` is needed here. I'm not sure whether we will have any options for synchronically lookup. So I would still +1 to `table.exec.async.xxx`.

Reply:
Or how about `table.exec.async-lookup.buffer-capacity` ?

Reply:
Sounds good.